src/pages/index.js

```diff
@@ -60,28 +107,122 @@ document.addEventListener("DOMContentLoaded", function () {
     addFormValidator.resetValidation();
   });
 
-  newCardList.renderItems();
+  constants.avatar.addEventListener("click", () => {
+    avatarPopup.open();
+    addFormValidator.resetValidation();
+  });
+
+  constants.avatar.addEventListener("mousemove", () => {
+    constants.avatarButton.classList.replace(
+      "profile__avatar-button_disabled",
+      "profile__avatar-button"
+    );
+  });
+});
+
+constants.avatar.addEventListener("mouseout", () => {
+  constants.avatarButton.classList.replace(
+    "profile__avatar-button",
+    "profile__avatar-button_disabled"
+  );
 });
 
 /* -------------------------------------------------------------------------- */
 /*                                  Functions                                 */
 /* -------------------------------------------------------------------------- */
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    (id) => handleDeleteClick(id),
+    (id) => handleLikeClick(id)
+  );
   return card.getView();
 }
 
-function handleImageClick(name, link) {
-  newPopupImage.open(name, link);
+function handleImageClick(cardData) {
+  newPopupImage.open(cardData);
+}
+
+function handleDeleteClick(card) {
+  popupAffirm.open();
+  popupAffirm.setSubmitAction(() => {
+    api
+      .deleteCards(card._id)
+      .then(() => {
+        popupAffirm.close();
+        card.handleDeleteCard();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleLikeClick(card) {
+  if (card.isLiked === true) {
+    api
+      .deleteCardLike(card._id)
+      .then((res) => {
+        card.isLiked = false;
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .putCardLike(card._id)
+      .then((res) => {
+        card.isLiked = true;
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
 }
 
 function handleProfileFormSubmit(formData) {
-  useInfo.setUserInfo(formData.title, formData.description);
-  editPopup.close();
+  constants.editProfileSubmitButton.textContent = "Saving...";
+  api
+    .updateProfileInfo(formData.title, formData.description)
+    .then((res) => {
+      useInfo.setUserInfo(res.name, res.about);
+      editPopup.close();
+      constants.editProfileSubmitButton.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.